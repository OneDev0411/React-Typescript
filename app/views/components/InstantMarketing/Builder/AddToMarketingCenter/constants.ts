export const CATEGORIES = [
  {
    name: 'Select a Category',
    value: 'none'
  },
  {
    name: 'Occasions',
    value: 'Birthday'
  },
  {
    name: 'Brand Campaigns',
    value: 'Brand'
  },
  {
    name: 'Holiday',
    value: 'OtherHoliday'
  },
  {
    name: 'New Agent',
    value: 'NewAgent'
  }
]

export const SAVED_TEMPLATE_VARIANT = 'SAVED_TEMPLATE'
