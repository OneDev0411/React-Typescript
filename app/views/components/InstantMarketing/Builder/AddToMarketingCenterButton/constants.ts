export const SAVED_TEMPLATE_VARIANT = 'SAVED_TEMPLATE'
