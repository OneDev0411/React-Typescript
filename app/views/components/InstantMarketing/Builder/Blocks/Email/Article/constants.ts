export const articleTopBlockName = 'rechat-article-image-top'
export const articleLeftBlockName = 'rechat-article-image-left'
export const articleRightBlockName = 'rechat-article-image-right'
