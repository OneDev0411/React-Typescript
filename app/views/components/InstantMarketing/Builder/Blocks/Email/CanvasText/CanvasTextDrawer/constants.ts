import { Konva, LabelConfig } from 'pikaso'

export interface CanvasTextProperties {
  container: Konva.LabelConfig
  text: Konva.TextConfig
  tag: Konva.TagConfig
  config?: LabelConfig
}

export const DefaultCanvasTextProperties: CanvasTextProperties = {
  container: {
    x: 0,
    y: 0
  },
  text: {
    text: 'Sample Text',
    fontSize: 40,
    fontFamily: 'Lato',
    lineHeight: 1.1,
    padding: 10,
    fill: '#000'
  },
  tag: {
    fill: 'transparent'
  }
}
