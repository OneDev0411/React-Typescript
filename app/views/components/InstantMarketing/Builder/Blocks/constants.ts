export const DEFAULT_CATEGORY = 'Rechat'
