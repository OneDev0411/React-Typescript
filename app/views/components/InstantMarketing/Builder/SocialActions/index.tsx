import React from 'react'

import { Button, useTheme, Theme } from '@material-ui/core'

import { SvgIcon } from 'components/SvgIcons/SvgIcon'

import { SocialNetworkShare } from '../types'

interface Props {
  networks: SocialNetworkShare[]
  onClick: (networkName: string) => void
}

export default function SocialActions({ networks, onClick }: Props) {
  const theme = useTheme<Theme>()

  return (
    <>
      {networks.map(network => (
        <Button
          key={network.name}
          variant="contained"
          color="primary"
          style={{ marginLeft: '0.5rem' }}
          onClick={() => onClick(network.name)}
        >
          <SvgIcon
            path={network.icon}
            style={{
              marginRight: theme.spacing(1)
            }}
          />
          Post to {network.name}
        </Button>
      ))}
    </>
  )
}
