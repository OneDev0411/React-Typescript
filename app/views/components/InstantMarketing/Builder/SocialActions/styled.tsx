import styled from 'styled-components'

import ActionButton from 'components/Button/ActionButton'

export const Button = styled(ActionButton)`
  &:not(:first-of-type) {
    margin-left: 0.5rem;
  }
`

export const Icon = styled.i`
  font-size: 1.5rem;
  margin: 0.5rem;
`
