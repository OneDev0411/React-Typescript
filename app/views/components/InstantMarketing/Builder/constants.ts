import { mdiFacebook, mdiInstagram, mdiLinkedin } from '@mdi/js'

import { SocialNetworkShare } from './types'

export const SOCIAL_NETWORKS: SocialNetworkShare[] = [
  {
    name: 'Instagram',
    icon: mdiInstagram
  },
  {
    name: 'Facebook',
    icon: mdiFacebook
  },
  {
    name: 'LinkedIn',
    icon: mdiLinkedin
  }
]

export const BASICS_BLOCK_CATEGORY = 'Basics'
export const LISTINGS_BLOCK_CATEGORY = 'Listings'
export const ARTICLES_BLOCK_CATEGORY = 'Articles'
export const AGENTS_BLOCK_CATEGORY = 'Agents'
export const MARKET_REPORTS_CATEGORY = 'Market Reports'
