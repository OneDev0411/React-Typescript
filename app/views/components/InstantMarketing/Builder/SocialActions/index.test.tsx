import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import { SOCIAL_NETWORKS } from '../constants'

import SocialActions from '.'

describe('MC SocialActions component', () => {
  it('renders', () => {
    const wrapper = shallow(
      <SocialActions onClick={() => null} networks={SOCIAL_NETWORKS} />
    )

    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders all needed buttons based on networks prop', () => {
    const wrapper = shallow(
      <SocialActions onClick={() => null} networks={SOCIAL_NETWORKS} />
    )

    expect(wrapper.children()).toHaveLength(SOCIAL_NETWORKS.length)

    SOCIAL_NETWORKS.map((network, index) => {
      const socialButton = wrapper.childAt(index)

      expect(socialButton.childAt(0)).toBeTruthy()
    })
  })
})
