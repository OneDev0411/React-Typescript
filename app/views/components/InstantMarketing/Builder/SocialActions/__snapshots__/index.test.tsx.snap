// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MC SocialActions component renders 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Button))
    color="primary"
    key="Instagram"
    onClick={[Function]}
    style={
      Object {
        "marginLeft": "0.5rem",
      }
    }
    variant="contained"
  >
    <SvgIcon
      path="M7.8,2H16.2C19.4,2 22,4.6 22,7.8V16.2A5.8,5.8 0 0,1 16.2,22H7.8C4.6,22 2,19.4 2,16.2V7.8A5.8,5.8 0 0,1 7.8,2M7.6,4A3.6,3.6 0 0,0 4,7.6V16.4C4,18.39 5.61,20 7.6,20H16.4A3.6,3.6 0 0,0 20,16.4V7.6C20,5.61 18.39,4 16.4,4H7.6M17.25,5.5A1.25,1.25 0 0,1 18.5,6.75A1.25,1.25 0 0,1 17.25,8A1.25,1.25 0 0,1 16,6.75A1.25,1.25 0 0,1 17.25,5.5M12,7A5,5 0 0,1 17,12A5,5 0 0,1 12,17A5,5 0 0,1 7,12A5,5 0 0,1 12,7M12,9A3,3 0 0,0 9,12A3,3 0 0,0 12,15A3,3 0 0,0 15,12A3,3 0 0,0 12,9Z"
      style={
        Object {
          "marginRight": 8,
        }
      }
    />
    Post to 
    Instagram
  </WithStyles(ForwardRef(Button))>
  <WithStyles(ForwardRef(Button))
    color="primary"
    key="Facebook"
    onClick={[Function]}
    style={
      Object {
        "marginLeft": "0.5rem",
      }
    }
    variant="contained"
  >
    <SvgIcon
      path="M12 2.04C6.5 2.04 2 6.53 2 12.06C2 17.06 5.66 21.21 10.44 21.96V14.96H7.9V12.06H10.44V9.85C10.44 7.34 11.93 5.96 14.22 5.96C15.31 5.96 16.45 6.15 16.45 6.15V8.62H15.19C13.95 8.62 13.56 9.39 13.56 10.18V12.06H16.34L15.89 14.96H13.56V21.96A10 10 0 0 0 22 12.06C22 6.53 17.5 2.04 12 2.04Z"
      style={
        Object {
          "marginRight": 8,
        }
      }
    />
    Post to 
    Facebook
  </WithStyles(ForwardRef(Button))>
  <WithStyles(ForwardRef(Button))
    color="primary"
    key="LinkedIn"
    onClick={[Function]}
    style={
      Object {
        "marginLeft": "0.5rem",
      }
    }
    variant="contained"
  >
    <SvgIcon
      path="M19 3A2 2 0 0 1 21 5V19A2 2 0 0 1 19 21H5A2 2 0 0 1 3 19V5A2 2 0 0 1 5 3H19M18.5 18.5V13.2A3.26 3.26 0 0 0 15.24 9.94C14.39 9.94 13.4 10.46 12.92 11.24V10.13H10.13V18.5H12.92V13.57C12.92 12.8 13.54 12.17 14.31 12.17A1.4 1.4 0 0 1 15.71 13.57V18.5H18.5M6.88 8.56A1.68 1.68 0 0 0 8.56 6.88C8.56 5.95 7.81 5.19 6.88 5.19A1.69 1.69 0 0 0 5.19 6.88C5.19 7.81 5.95 8.56 6.88 8.56M8.27 18.5V10.13H5.5V18.5H8.27Z"
      style={
        Object {
          "marginRight": 8,
        }
      }
    />
    Post to 
    LinkedIn
  </WithStyles(ForwardRef(Button))>
</Fragment>
`;
